internals/secrethub/service_aws_init.go

```diff
@@ -0,0 +1,95 @@
+package secrethub
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/errio"
+)
+
+// Errors
+var (
+	ErrInvalidAWSRegion = errMain.Code("invalid_region").Error("invalid AWS region")
+	ErrRoleAlreadyTaken = errMain.Code("role_taken").Error("a service using that IAM role already exists")
+)
+
+// ServiceAWSInitCommand initializes a service for AWS.
+type ServiceAWSInitCommand struct {
+	description string
+	path        api.DirPath
+	kmsKeyID    string
+	role        string
+	region      string
+	permission  api.Permission
+	io          ui.IO
+	newClient   newClientFunc
+}
+
+// NewServiceAWSInitCommand creates a new ServiceAWSInitCommand.
+func NewServiceAWSInitCommand(io ui.IO, newClient newClientFunc) *ServiceAWSInitCommand {
+	return &ServiceAWSInitCommand{
+		io:        io,
+		newClient: newClient,
+	}
+}
+
+// Run initializes an AWS service.
+func (cmd *ServiceAWSInitCommand) Run() error {
+	repo := cmd.path.GetRepoPath()
+
+	client, err := cmd.newClient()
+	if err != nil {
+		return err
+	}
+
+	cfg := aws.NewConfig()
+	if cmd.region != "" {
+		_, ok := endpoints.AwsPartition().Regions()[cmd.region]
+		if !ok {
+			return ErrInvalidAWSRegion
+		}
+		cfg = cfg.WithRegion(cmd.region)
+	}
+
+	service, err := client.Services().AWS().Create(repo.Value(), cmd.description, cmd.kmsKeyID, cmd.role, cfg)
```

Comment:
Why do we pass that config to the AWS call and not create a new client that is decorated with the AWS client, which allows us to make that function call without the 'one-time setup' arguments?